lookout/style/format/benchmarks/compare_quality_reports.py

```diff
@@ -0,0 +1,85 @@
+from typing import Optional, Union
+
+import pandas
+from tabulate import tabulate
+
+from lookout.style.format.benchmarks.top_repos_quality import FLOAT_PRECISION
+
+
+def _convert_cell_to_value(cell: str) -> Union[int, float, str]:
+    for type_ in (int, float):
+        try:
+            return type_(cell.strip())
+        except ValueError:
+            pass
+    cell = cell.strip()
+    return cell if cell else float("nan")
+
+
+def _quality_report_table_to_df(report_table_path: str) -> pandas.DataFrame:
+    table = []
+    separator_set = frozenset(":|-")
+    with open(report_table_path) as f:
+        for line in f:
+            if set(line.strip()) <= separator_set:
+                continue
+            cells = line.strip().split("|")
+            if not cells[0]:
+                cells = cells[1:]
+            if not cells[-1]:
+                cells = cells[:-1]
+            values = [_convert_cell_to_value(c) for c in cells]
+            table.append(values)
+    return pandas.DataFrame(table[1:], columns=table[0])
+
+
+def compare_quality_reports(base: str, new: str, output: Optional[str] = None) -> None:
+    """
+    Print a table with metric difference between the reports.
+
+    If output argument is provided comparison result is saved to new file.
+
+    :param base: Baseline report. Usually the latest report from ./report/ directory.
+    :param new: New report. Usually It is a report generated for master or any local \
+                       change you did and want to validate.
+    :param output: The result will be saved to this File path if provided.
+    """
+    base_report = _quality_report_table_to_df(base)
+    new_report = _quality_report_table_to_df(new)
+    base_report.set_index("repo", inplace=True)
+    new_report.set_index("repo", inplace=True)
+
+    delta = new_report - base_report
+    delta = delta.reindex(index=list(base_report.index))
+    formats = {
+        "precision": FLOAT_PRECISION,
+        "recall": FLOAT_PRECISION,
+        "full_recall": FLOAT_PRECISION,
+        "f1": FLOAT_PRECISION,
+        "full_f1": FLOAT_PRECISION,
+        "ppcr": FLOAT_PRECISION,
+        "support": "7g",
+        "full_support": "7g",
+        "Rules Number": "4g",
+        "Average Rule Len": ".1f",
+    }
+    for field in delta.columns:
+        delta[field] = new_report[field].map(lambda x: ("%%%s" % formats[field] % x)) + \
+                       delta[field].map(lambda x: " (%%+%s)" % formats[field] % x)
+
+    delta = delta.fillna("")
+    delta.loc[["weighted average"],
+              ["support", "full_support", "Rules Number", "Average Rule Len"]] = ""
+
+    res = "\n# Report comparison\n%s\nvs\n%s\n\n" % (new, base)
+    res += tabulate(delta, tablefmt="pipe", headers="keys", stralign="right")
+
+    print(res)
```

Comment:
The output should be single, either a file or stdout. I suggest you to output to a file object, and it will be the caller's responsibility to provide it.